openedx/features/course_experience/utils.py

```diff
@@ -226,3 +230,45 @@ def is_block_structure_complete_for_assignments(block_data, block_key):
     scored = has_score and (weight is None or weight > 0)
 
     return complete or not graded or not scored
+
+
+def reset_deadlines_for_course(request, course_key, research_event_data={}):  # lint-amnesty, pylint: disable=dangerous-default-value
+    """
+    Set the start_date of a schedule to today, which in turn will adjust due dates for
+    sequentials belonging to a self paced course
+
+    Args:
+        request (Request): The request object
+        course_key (str): The course key
+        research_event_data (dict): Any data that should be included in the research tracking event
+            Example: sending the location of where the reset deadlines banner (i.e. outline-tab)
+    """
+    from lms.djangoapps.course_api.api import course_detail  # lint-amnesty, pylint: disable=import-outside-toplevel
```

Comment:
I suggest moving this to a new file openedx/features/course_experience/api/v1/utils.py to prevent this problem with circular imports.